cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/DefaultWorkflowConfigurer.java

```diff
@@ -112,8 +139,7 @@ private WorkflowNode createNodeWithId(WorkflowNode node) {
     WorkflowNode nodeWithId = null;
     switch (node.getType()) {
       case ACTION:
-        nodeWithId = createActionNodeWithId(node);
-        break;
+        return node;
```

Comment:
I don't understand this method. It adds an id to the node? So now actions don't have an id any more, but for and conditions do?